interfaces/devdoc/sync_requirements.md

```diff
@@ -0,0 +1,42 @@
+# sync
+================
+
+## Overview
+
+`sync` provides platform-independent synchronization primitives.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout);
+MOCKABLE_FUNCTION(, void, wake_by_address_all, void*, address);
+MOCKABLE_FUNCTION(, void, wake_by_address_single, void*, address);
+```
+
+## wait_on_address
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout)
+```
+
+**SRS_SYNC_43_001: [** `wait_on_address` shall atomically compare `*address` and `compare_address`.**]**
+
+**SRS_SYNC_43_002: [** `wait_on_address` shall immediately return false if `*address` is not equal to `*compare_address`.**]**
+
+**SRS_SYNC_43_003: [** `wait_on_address` shall wait until another thread in the same process signals at `address` using `wake_by_address_single` or `wake_by_address_all`, or `timeout` milliseconds pass, whichever comes first. **]**

```

Comment:
>wait until another thread in the same process signals at [](start = 47, length = 56)

Should probably specify that this depends on `compare_address`. Also it can be worken spuriously (at least on Windows, I assume on Linux as well). Maybe just say "shall wait until the value at `address` changes to something other than `compare_address` by means of an operating system call (`address` and `compare_address` are not guaranteed to be different after this returns)."

Reply:
I'd probably separate out the timeout part to a separate spec too, because this one seems long (If `timeout` milliseconds elapse then `wait_on_address` shall...).

---
In reply to: [441222237](https://github.com/Azure/azure-c-pal/pull/5#discussion_r441222237) [](ancestors = 441222237)